org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService.xtend

```diff
@@ -47,19 +48,22 @@ class RenameService implements IRenameService {
 		workspaceManager.doRead(uri) [ document, resource | 
 			val projectManager = workspaceManager.getProjectManager(uri)
 			val resourceSet = projectManager.createNewResourceSet(projectManager.indexState.resourceDescriptions)
+			resourceSet.loadOptions.put(ResourceDescriptionsProvider.LIVE_SCOPE, true)
 			val offset = document.getOffSet(renameParams.position)
 			val workspaceEdit = new WorkspaceEdit
 			val xtextResource = resourceSet.getResource(resource.URI, true)
 			if (xtextResource instanceof XtextResource) {
-				val element = xtextResource.resolveElementAt(offset)
-				if (element === null)
+				val element = EcoreUtil.resolve(xtextResource.resolveElementAt(offset), resourceSet)
```

Comment:
`resolveElementAt` should yield a resolved EObject. Why is the failsafe seemingly redundant call here necessary?